libgnucash/engine/engine.i

```diff
@@ -65,6 +65,7 @@ SCM scm_init_sw_engine_module (void);
 
 %import "base-typemaps.i"
 
+GLIST_HELPER_INOUT(DatesList, SWIGTYPE_p_Time64);
```

Comment:
Why did you tell it that it contains a `SWIGTYPE_p_Time64` instead of a `time64`?

Reply:
This is what happens with `time64`:

```
libgnucash/engine/swig-engine.c: In function ‘_wrap_xaccAccountGetReconcileStatementDates’:
libgnucash/engine/swig-engine.c:7730:9: error: expected expression before ‘time64’
 7730 |         time64, 0), list);
      |         ^~~~~~
libgnucash/engine/swig-engine.c:776:40: note: in definition of macro ‘SWIG_NewPointerObj’
  776 |   SWIG_Guile_NewPointerObj((void*)ptr, type, owner)
      |                                        ^~~~
libgnucash/engine/swig-engine.c:776:3: error: too few arguments to function ‘SWIG_Guile_NewPointerObj’
  776 |   SWIG_Guile_NewPointerObj((void*)ptr, type, owner)
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
libgnucash/engine/swig-engine.c:7729:21: note: in expansion of macro ‘SWIG_NewPointerObj’
 7729 |     list = scm_cons(SWIG_NewPointerObj(node->data,
      |                     ^~~~~~~~~~~~~~~~~~
libgnucash/engine/swig-engine.c:831:1: note: declared here
  831 | SWIG_Guile_NewPointerObj(void *ptr, swig_type_info *type, int owner)
      | ^~~~~~~~~~~~~~~~~~~~~~~~
[20/152] Building CXX object libgnucash/engine/CMakeFiles/gncmod-engine.dir/Account.cpp.o
ninja: build stopped: subcommand failed.
```

Reply:
K. The immediate cause of the error is that you called `GLIST_HELPER_INOUT` before the `%include "gnc-date.h"` on line 144 so the compiler hadn't yet hit the `typedef` of `time64` and didn't know it's a type. The next problem is that `GLIST_HELPER_INOUT` is for GLists of pointers. That won't work for int64_t. You need a `typemap(out) DatesList*`.

```
%typemap(out) DatesList*
{
    SCM list = SCM_EOL;
    GList* node;
    for (node = $1; node; node = g_list_next(node))
        list = scm_cons(scm_from_int64(node->data), list);

    $result = scm_list_reverse(list);
}
```